gulp/docker/run-functional-test-via-command-line.js

```diff
@@ -0,0 +1,48 @@
+const tmp          = require('tmp');
+const copy         = require('recursive-copy');
+const path         = require('path');
+const childProcess = require('child_process');
+const os           = require('os');
+const osFamily     = require('os-family');
+
+function correctPathOnWindows (tmpDir) {
+    const parsedPath             = path.parse(tmpDir);
+    const newRoot                = '//' + parsedPath.root.replace(':', '');
+    const modifiedTmpDir         = tmpDir.replace(parsedPath.root, newRoot);
+    const modifiedTmpDirSegments = modifiedTmpDir.split(path.sep);
+    const userNameTmpDirPart     = modifiedTmpDirSegments[2];
+
+    if (userNameTmpDirPart.includes('~')) {
+        // NOTE: Docker cannot resolve paths with `~` symbol on Windows operation system
+        const userInfo = os.userInfo();
+
+        modifiedTmpDirSegments[2] = userInfo.username;
+    }
+
+    return modifiedTmpDirSegments.join(path.posix.sep);
+}
+
+function setAccessPrivileges (dir) {
+    const cmd = `chmod -R 777 ${dir}`;
+
+    childProcess.execSync(cmd, { stdio: 'inherit', env: process.env });
+}
+
+module.exports = function runFunctionalTestViaCommandLine (publishRepository, packageInfo) {
+    tmp.setGracefulCleanup();
+
+    const tmpDir   = tmp.dirSync();
+    const testsDir = path.join(__dirname, '../../test/docker/testcafe-fixtures');
+
+    return copy(testsDir, tmpDir.name).then(() => {
+        const resultTmpDirPath = osFamily.win ? correctPathOnWindows(tmpDir.name) : tmpDir.name;
+
+        if (osFamily.linux)
+            setAccessPrivileges(resultTmpDirPath);
+
+        const cmd = `docker run -v ${resultTmpDirPath}:/tests -i ${publishRepository}:${packageInfo.version} chromium /tests/*.js`;
```

Comment:
We can safely remove the container after its usage is done (via [--rm](https://docs.docker.com/engine/reference/run/#clean-up---rm)) and replace the `chromium` browser alias with `all`.

Reply:
This test covers test running on docker via command-line interface. It's necessary to use all installed browsers.
However, removing the containers on task end is a good idea.